### PR TITLE
Add remaining direct messages API end points

### PR DESCRIPTION
**Pull request by waseem**
Add ExTwitter.direct_messages(:sent, options // [])

Add ExTwitter.destroy_direct_message/1 and ExTwitter.destroy_direct_message/2

Add ExTwitter.new_direct_message/2

### CODE REVIEW

**Comment by parroty** [comment]
Thanks. I'll be putting some comments inline, and please let me know how you think.

File: lib/extwitter.ex [hunk 1]
```diff
@@ -582,6 +582,60 @@ defmodule ExTwitter do
...
+  GET direct_messages/sent
+
+  ## Examples
+
+      ExTwitter.direct_messages(:sent, count: 1)
+
+  ## Reference
+  https://dev.twitter.com/rest/reference/get/direct_messages/sent
+  """
+  @spec direct_messages(:sent, Keyword.t) :: [ExTwitter.Model.DirectMessage.t]
```
Comment (parroty) [inline lib/extwitter.ex]: I think having the `sent` as argument sounds tricky as it's just different functionality. How about just having separate signature like `direct_messages_sent`?

Comment (parroty) [inline lib/extwitter.ex]: Also, I think we can have zero argument version for `direct_messages(:sent)` as all the options are `optional` one according to the API doc.

Comment (waseem) [inline lib/extwitter.ex]: That sounds fine.

**Comment by waseem** [comment]
@parroty I made those changes. Check those out. :smiley:

**Comment by parroty** [comment]
Thanks!

**Pull request merged**